sopel/tools/memories.py

```diff
@@ -13,11 +13,27 @@
 from .identifiers import Identifier
 
 if TYPE_CHECKING:
-    from typing import Callable, Optional
+    from collections.abc import Generator, Mapping
+    from typing import Any, Callable, Optional, Union
 
     IdentifierFactory = Callable[[str], Identifier]
 
 
+class _NO_DEFAULT:
```

Comment:
If it makes you feel better, the stdlib does it in a whole bunch of places, but it's not really a big deal in this case since no instances are ever constructed, and it does give us a convenient place to hang a docstring.

Reply:
I think it's OK. If someone find a better solution in the future, it's only a PR away. I'm OK with that!